src/Data/Food/Builder/Query.elm

```diff
@@ -92,25 +93,29 @@ carrotCake =
           , mass = Mass.grams 120
           , variant = Default
           , country = Nothing
+            , byPlane = False
           }
         , { id = Ingredient.idFromString "wheat"
           , name = "Blé tendre"
           , mass = Mass.grams 140
           , variant = Default
           , country = Nothing
+            , byPlane = False
```

Comment:
The same way we have `Maybe Country` helping identifying when no option is explicitely passed in an query (especially with the API), we may probably want a `Maybe Bool` here. Also that would keep things consistent.

I understand it's complicated. Basically, `Maybe a` will help you understand better if the user has explicitly overridden a default or not; `Nothing` means the query will rely on default value, whatever it is: it doesn't know about it. `Just "x"` means the user have explicitely set the value to `"x"`. So when we serialize the query, we can distinguish if the user has overridden a default or not. If we enforce a value being always present, you can't easily know if the user has overridden it or not. And if we ever change the default values, thery're not hardcoded in serialized/saved queries (eg. bookmarks or api call urls).

I bet it's even more confusing now. Available to chat about it.

Reply:
mmmm, I'm having some "impossible states" vibes: in theory i could have a mango (`defaultOrigin` is `OutOfEuropeAndMaghrebByPlane`) that has `country = Nothing` (so should use its default origin), and `byPlane = False` (or even `byPlane = Just False`).

I guess this means there's an issue with data modeling... any ideas?

Reply:
I'll go with it for now, we'll see if this needs a refactor in the future. For now the validation of correct data is done in `Recipe.ingredientFromQuery` and I added a test to make sure this is done.